messages.go

```diff
@@ -190,22 +155,21 @@ func (kp KeyPackage) Verify() bool {
 	return kp.Credential.Scheme().Verify(kp.Credential.PublicKey(), tbs, kp.Signature.Data)
 }
 
-func NewKeyPackage(suite CipherSuite, cred *Credential) (*KeyPackage, error) {
-	initKey, err := suite.hpke().Generate()
+func NewKeyPackageWithSecret(suite CipherSuite, initSecret []byte, cred *Credential, sigPriv SignaturePrivateKey) (*KeyPackage, error) {
+	initPriv, err := suite.hpke().Derive(initSecret)
 	if err != nil {
 		return nil, err
 	}
 
-	return NewKeyPackageWithInitKey(suite, initKey, cred)
+	return NewKeyPackageWithInitKey(suite, initPriv.PublicKey, cred, sigPriv)
```

Comment:
may be returning initPriv here will save a Derive() call at the app layer ?

Reply:
Actually, I'm increasingly thinking that the application should not think about init keys, just secrets.  So the application would provide a secret and a signature private key, and the MLS stack would handle the Derive() call.  That can be a follow-on, though.